src/middlewared/middlewared/plugins/filesystem.py

```diff
@@ -878,6 +879,21 @@ def setacl(self, job, data):
         self._winacl(data['path'], 'clone', uid, gid, options)
         job.set_progress(100, 'Finished setting ACL.')
 
+    @private
+    async def path_is_encrypted(self, path):
+        p = Path(path)
+        if not p.is_absolute():
+            raise CallError(f"[{path}] is not an absolute path.", errno.EINVAL)
+
+        rp = p.resolve()
+        our_pool = p.parts[2]
+
+        for ds in await self.middleware.call('pool.dataset.query', [('pool', '=', our_pool)]):
```

Comment:
This will loop through all datasets, is there some way we can restrict to the path we are checking?